src/scene/assetmanager.cpp

```diff
@@ -24,106 +24,751 @@
 
 #include <openspace/scene/assetmanager.h>
 
-#include <openspace/engine/globals.h>
-#include <openspace/scene/assetloader.h>
-#include <openspace/scene/profile.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/scene/asset.h>
 #include <openspace/scripting/lualibrary.h>
-#include <openspace/util/synchronizationwatcher.h>
-#include <ghoul/filesystem/file.h>
+#include <ghoul/filesystem/filesystem.h>
 #include <ghoul/logging/logmanager.h>
-#include <ghoul/misc/exception.h>
-#include <ghoul/misc/profiling.h>
 
 #include "assetmanager_lua.inl"
 
+namespace {
+    constexpr const char* _loggerCat = "AssetManager";
+
+    constexpr const char* AssetGlobalVariableName = "asset";
+
+    constexpr const char* ExportsTableName = "_exports";
+    constexpr const char* AssetTableName = "_asset";
+
+    enum class PathType {
+        RelativeToAsset,
+        RelativeToAssetRoot,
+        Absolute
+    };
+
+    PathType classifyPath(const std::string& path) {
+        if (path.size() > 2 && path[0] == '.' && path[1] == '/') {
+            return PathType::RelativeToAsset;
+        }
+        if (path.size() > 3 && path[0] == '.' && path[1] == '.' && path[2] == '/') {
+            return PathType::RelativeToAsset;
+        }
+        if (path.size() > 3 && path[1] == ':' && (path[2] == '\\' || path[2] == '/')) {
+            return PathType::Absolute;
+        }
+        if (path.size() > 1 && (path[0] == '\\' || path[0] == '/')) {
+            return PathType::Absolute;
+        }
+        return PathType::RelativeToAssetRoot;
+    }
+
+    struct [[codegen::Dictionary(AssetMeta)]] Parameters {
+        std::optional<std::string> name;
```

Comment:
No docs? :) 

I guess we already have all that information in the files for the assets, but those are not actually exposed to the users anywhere. However, maybe we don't need to, as it should probably live on a wiki page somewhere

Reply:
Added documentation